stackstorm-nexus3/actions/lib/base.py

```diff
@@ -0,0 +1,178 @@
+import sys
+from exception import *
+from nexuscli.repository import Repository
+from nexuscli.nexus_client import NexusClient
+from nexuscli.exception import *
+
+from st2common.runners.base_action import Action
+
+__all__ = [
+    'BaseAction'
+]
+
+REQUIRED_FIELDS = ['url', 'user', 'password']
+
+
+class BaseAction(Action):
+
+    def __init__(self, config):
+        super(BaseAction, self).__init__(config)
+        self.nexus_profiles = self.config.get('profiles', {})
+        self.nexus_client = None
+        self.dial_config = None
+
+    def validate_config(self, c_profile):
+        """ Validate connection configuration
+
+        """
+        validation_errors = []
+        is_valid = True
+        for field in REQUIRED_FIELDS:
+            if c_profile.get(field, None) == None:
+                validation_errors.push("%s if missing" % field)
+                is_valid = False
+
+        return (is_valid, validation_errors)
+
+    def get_connection_defaults(self):
+        """ default properties
+
+        """
+        return {
+            "url": self.config.get('url', None),
+            "verify": self.config.get('verify', True),
+            "user": self.config.get('user', None),
+            "password": self.config.get('password', None)
+        }
+
+    def init_config(self, profile=None):
+        """ Get nexus3 server connection string
+
+        """
+        self.dial_config = self.get_connection_defaults()
+        if profile is not None:
+            profile = profile.strip()
+            if len(profile) == 0:
+                profile = None
+
+        # use 'default_profile' if 'config_profile' is empty
+        if profile is None:
+            profile = self.config.get('default_profile', None)
+
+        # Use defaults options, if neither config_profile nor default_profile is provided
+        if profile is None:
```

Comment:
This check can be removed and the `self.logger.info` line right below can be moved into the body at line 61.